### ISSUE TITLE
Phaser Baseband AA Filter Specification

### ISSUE THREAD

**Issue by Jonah-Foley** [sinara-hw/Phaser#158]
Hi, i'm just trying to get a handle on what the actual specification for the anti-aliasing filter used on Phaser (basedband) is. The [Phaser classic gateware](https://github.com/quartiq/phaser/wiki) provides a block diagram of the system where it quotes the AA filter to have a 340MHz cut-off frequency, however upon measuring this the following spectrum is seen:

![image](https://github.com/sinara-hw/Phaser/assets/152205416/55ec56aa-18f1-4329-9319-ccd314b35f89)
 
This shows the frequency response in the range [100MHz, 500MHz] and at 300MHz we see we are about 35dB(m) down from the passband (which is correctly at ~0dB), with the actual cut-off around 200MHz. 

I have looked through #4, #119. #130 and so on to try to find a clear definition of the spec but its hard to tell what decisions were made. In addition i'm having a hard time telling why the quoted bandwidth is 340MHz in the wiki if the actual bandwidth is significantly lower. Any clarity on this issue is appreciated.

**Comment by gkasprow** [comment]
How did you measure the filter frequency response?

**Comment by Jonah-Foley** [comment]
Hi Greg, the experimental setup was as follows:

- Configure the DAC34H84 to use the NCO, disable all other coarse grained mixers.
- (Via an ARTIQ script) every 100ms increment the NCO frequency over the range 100 - 500 MHz, driving the DAC input with a constant value (0x7FFF_FFFF) via the `dac_test_data` register.
- Connect the Phaser RF output to a spectrum analyzer configured to max hold with a 20ms sampling period per increment.

The result is the trace above.

We now have a solution for this issue and my colleague will be providing full details on Tuesday 7th May.

**Comment by nqtim** [comment]
It looks like the issue is to do with scaling the capacitor values for a differential filter.  In [this ticket](https://github.com/sinara-hw/Phaser/issues/119#issuecomment-1243415558) we can see the motivations for the original filter design.  However, when scaling the single-ended filter for differential, the capacitors were doubled, whereas they actually need to be halved.

We also accounted for the 5pF parasitic capacitance on each DAC output pin (according to the datasheet) which adds an extra 2.5pF equivalent differential, so we reduced the first capacitor from 3pF to 0.5pF.

Here are the values we used:
![image](https://github.com/sinara-hw/Phaser/assets/108869873/5df23f92-2cfe-4a15-bff6-068428a21aac)

We modified a baseband board to these values.  Here is the measured result to 500MHz with these values (ignore the bumps at DC and 500MHz):
![Phaser_mod_AA_filter_response](https://github.com/sinara-hw/Phaser/assets/108869873/916298bd-d79f-4042-a73a-a8ae447f657e)



Probably a little fine-tuning or more detailed accounting of parasitics would help to level out the passband, but we didn't bother since this is good enough for us.

**Comment by jordens** [comment]
Ah sure.
Thanks for the debugging!

**Comment by gkasprow** [comment]
Right! We have two capacitors connected in series with virtual ground between them. My bad!

**Comment by kaolpr** [comment]
We've modified one of our devices with `C1=0.5p`, `C2=4.7p`, and `C3=3p` and obtained a similar result:
![image](https://github.com/sinara-hw/Phaser/assets/9166701/64bd2a14-7422-4cbd-a156-ad109e69bf57)
The question is - is the bandwidth characteristic obtained by @nqtim satisfactory, or do we need some more work on that?
